File: hrms-frontend/src/services/trainingService.js
// src/services/trainingService.js
import api from '../api/axiosConfig';

const trainingService = {
  // Programs
  getAllPrograms: () => api.get('/training/programs'),
  getOpenPrograms: (empId) =>
    api.get(`/training/programs/open${empId ? `?employeeId=${empId}` : ''}`),
  getProgram: (id, empId) =>
    api.get(`/training/programs/${id}${empId ? `?employeeId=${empId}` : ''}`),
  createProgram: (data) => api.post('/training/programs', data),
  updateProgram: (id, data) => api.put(`/training/programs/${id}`, data),
  deleteProgram: (id) => api.delete(`/training/programs/${id}`),
  updateStatus: (id, status) =>
    api.patch(`/training/programs/${id}/status?status=${status}`),

  // Enrollments
  enroll: (programId, employeeId) =>
    api.post('/training/enroll', { programId, employeeId }),
  getMyEnrollments: (empId) =>
    api.get(`/training/enrollments/employee/${empId}`),
  getEnrollmentsByProgram: (programId) =>
    api.get(`/training/enrollments/program/${programId}`),
  markComplete: (id, data) =>
    api.patch(`/training/enrollments/${id}/complete`, data),
  submitFeedback: (id, data) =>
    api.patch(`/training/enrollments/${id}/feedback`, data),
  dropEnrollment: (id) => api.delete(`/training/enrollments/${id}`),

  getStatistics: () => api.get('/training/statistics'),
  getEmployeeStats: (empId) => api.get(`/training/statistics/employee/${empId}`),
};

export default trainingService;

File: hrms-backend/src/main/java/com/hrms/service/TrainingService.java

//service/TrainingService.java
package com.hrms.service;

import com.hrms.dto.response.TrainingEnrollmentResponse;
import com.hrms.dto.response.TrainingProgramResponse;
import com.hrms.model.TrainingProgram;
import com.hrms.model.TrainingEnrollment;

import java.util.List;
import java.util.Map;

public interface TrainingService {
 // Programs
 TrainingProgramResponse createProgram(TrainingProgram program);
 TrainingProgramResponse updateProgram(Long id, TrainingProgram program);
 TrainingProgramResponse getProgram(Long id, Long employeeId);
 List<TrainingProgramResponse> getAllPrograms();
 List<TrainingProgramResponse> getProgramsByCategory(String category);
 List<TrainingProgramResponse> getOpenPrograms(Long employeeId);
 void deleteProgram(Long id);
 TrainingProgramResponse updateStatus(Long id, String status);

 // Enrollments
 TrainingEnrollmentResponse enroll(Long programId, Long employeeId);
 TrainingEnrollmentResponse updateEnrollment(Long id, TrainingEnrollment enrollment);
 TrainingEnrollmentResponse markComplete(Long id, java.math.BigDecimal score,
                                           String grade, String certificateUrl);
 TrainingEnrollmentResponse submitFeedback(Long id, Integer rating,
                                             String feedback);
 List<TrainingEnrollmentResponse> getMyEnrollments(Long employeeId);
 List<TrainingEnrollmentResponse> getEnrollmentsByProgram(Long programId);
 void dropEnrollment(Long id);

 // Statistics
 Map<String, Object> getStatistics();
 Map<String, Object> getEmployeeLearningStats(Long employeeId);
}